docs/reference/aggregations/metrics/top-metrics-aggregation.asciidoc

```diff
@@ -0,0 +1,281 @@
+        "field": "ip"
+      },
+      "aggs": {
+        "tm": {
+          "top_metrics": {
+            "metric": {"field": "v"},
+            "sort": {"date": "desc"}
+          }
+        }
+      }
+    }
+  }
+}
+----
+
+Which returns:
+
+[source,js]
+----
+{
+  "aggregations": {
+    "ip": {
+      "buckets": [
+        {
+          "key": "192.168.0.1",
+          "doc_count": 2,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 2.0 } } ]
+          }
+        },
+        {
+          "key": "192.168.0.2",
+          "doc_count": 1,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 3.0 } } ]
+          }
+        }
+      ],
+      "doc_count_error_upper_bound": 0,
+      "sum_other_doc_count": 0
+    }
+  }
+}
+----
+// TESTRESPONSE
+
+Unlike `top_hits`, you can sort buckets by the results of this metric:
+
+[source,console]
+----
+POST /node/_search?filter_path=aggregations
+{
+  "aggs": {
+    "ip": {
+      "terms": {
+        "field": "ip",
+        "order": {"tm.v": "desc"}
+      },
+      "aggs": {
+        "tm": {
+          "top_metrics": {
+            "metric": {"field": "v"},
+            "sort": {"date": "desc"}
+          }
+        }
+      }
+    }
+  }
+}
+----
+// TEST[continued]
+
+Which returns:
+
+[source,js]
+----
+{
+  "aggregations": {
+    "ip": {
+      "buckets": [
+        {
+          "key": "192.168.0.2",
+          "doc_count": 1,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 3.0 } } ]
+          }
+        },
+        {
+          "key": "192.168.0.1",
+          "doc_count": 2,
+          "tm": {
+            "top": [ {"sort": ["2020-01-01T02:01:01.000Z"], "metrics": {"v": 2.0 } } ]
+          }
+        }
+      ],
+      "doc_count_error_upper_bound": 0,
+      "sum_other_doc_count": 0
+    }
+  }
+}
+----
+// TESTRESPONSE
+
+===== Mixed sort types
+
+Sorting `top_metrics` by a field that has different types across different
+indices producs somewhat suprising results: floating point fields are
+always sorted *before* whole number fields.
+
+[source,console,id=search-aggregations-metrics-top-metrics-mixed-sort]
+----
+POST /test/_bulk?refresh
+{"index": {"_index": "test1"}}
+{"s": 1, "v": 3.1415}
+{"index": {"_index": "test1"}}
+{"s": 2, "v": 1}
+{"index": {"_index": "test2"}}
+{"s": 3.1, "v": 2.71828}
+POST /test*/_search?filter_path=aggregations
+{
+  "aggs": {
+    "tm": {
+      "top_metrics": {
+        "metric": {"field": "v"},
+        "sort": {"s": "asc"}
+      }
+    }
+  }
+}
+----
+
+Which returns:
+
+[source,js]
+----
+{
+  "aggregations": {
+    "tm": {
+      "top": [ {"sort": [3.0999999046325684], "metrics": {"v": 2.718280076980591 } } ]
+    }
+  }
+}
+----
+// TESTRESPONSE
+
+While this is better than an error it *probably* isn't what you were going for.
+You can explictly cast the floating point fields to integers or the way around
```

Comment:
When mixing floats and integers, wouldn't users more often want to cast integers to floats, which is a lossless conversion for small integers?

Reply:
Yeah, probably. I'll swap the example around.